flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointsCleaner.java

```diff
@@ -43,13 +49,19 @@ public class CheckpointsCleaner implements Serializable, AutoCloseableAsync {
     private static final Logger LOG = LoggerFactory.getLogger(CheckpointsCleaner.class);
     private static final long serialVersionUID = 2545865801947537790L;
 
-    @GuardedBy("this")
+    private final Object lock = new Object();
```

Comment:
There are now sections synchronized on this field and on `this`.
Should it be a single object (either `this` or `lock`)?